docn/cime_config/buildnml

```diff
@@ -142,9 +133,9 @@ def buildnml(case, caseroot, compname):
 
     # NOTE: User definition *replaces* existing definition.
     files = Files(comp_interface="nuopc")
-    definition_file = [files.get_value("NAMELIST_DEFINITION_FILE", {"component":"docn"})]
+    definition_file = [files.get_value("NAMELIST_DEFINITION_FILE", {"component":compname})]
```

Comment:
If this change to compname makes this subroutine generic perhaps we should move it to a shared location and remove it from individual components?